nwb_conversion_tools/datainterfaces/behavior/deeplabcut/deeplabcutdatainterface.py

```diff
@@ -0,0 +1,143 @@
+        # Fetch the corresponding metadata pickle file
+        paf_graph = []
+        filename, _ = os.path.splitext(self.source_data["dlc_file_path"])
+        for i, c in enumerate(filename[::-1]):
+            if c.isnumeric():
+                break
+        if i > 0:
+            filename = filename[:-i]
+        metadata_file = filename + "_meta.pickle"
+        if os.path.isfile(metadata_file):
+            with open(metadata_file, "rb") as file:
+                metadata = pickle.load(file)
+            test_cfg = metadata["data"]["DLC-model-config file"]
+            paf_graph = test_cfg.get("partaffinityfield_graph", [])
+            if paf_graph:
+                paf_inds = test_cfg.get("paf_best")
+                if paf_inds is not None:
+                    paf_graph = [paf_graph[i] for i in paf_inds]
+        else:
+            warnings.warn("Metadata not found...")
+
+        for video_path, params in self._config_file["video_sets"].items():
+            if self._derived_metadata["vidname"] in video_path:
+                video = video_path, params["crop"]
+                break
+
+        if video is None:
+            warnings.warn(f"The video file corresponding to {self.source_data['dlc_file_path']} could not be found...")
+            video = "fake_path", "0, 0, 0, 0"
+
+            timestamps = df.index.tolist()
+        else:
+            timestamps = get_movie_timestamps(video[0])
+
+        for animal, df_ in df.groupby(level="individuals", axis=1):
+            pose_estimation_series = []
+            for kpt, xyp in df_.groupby(level="bodyparts", axis=1, sort=False):
+                data = xyp.to_numpy()
+
+                pes = PoseEstimationSeries(
+                    name=f"{animal}_{kpt}",
+                    description=f"Keypoint {kpt} from individual {animal}.",
+                    data=data[:, :2],
+                    unit="pixels",
+                    reference_frame="(0,0) corresponds to the bottom left corner of the video.",
+                    timestamps=timestamps,
+                    confidence=data[:, 2],
+                    confidence_definition="Softmax output of the deep neural network.",
+                )
+                pose_estimation_series.append(pes)
+
+            pe = PoseEstimation(
+                pose_estimation_series=pose_estimation_series,
+                description="2D keypoint coordinates estimated using DeepLabCut.",
+                original_videos=[video[0]],
+                dimensions=[list(map(int, video[1].split(",")))[1::2]],
+                scorer=self._derived_metadata["scorer"],
+                source_software="DeepLabCut",
+                source_software_version=__version__,
+                nodes=[pes.name for pes in pose_estimation_series],
+                edges=paf_graph,
+            )
+
+            get_module(nwbfile=nwbfile, name="behavior", description="processed behavioral data").add(pe)
```

Comment:
It looks like what you've done here is copy/paste [a large section of code in DLC2NWB](https://github.com/DeepLabCut/DLC2NWB/blob/5c704446a41f7fd6dfe9bcf7ceeedde69ec03717/dlc2nwb/utils.py#L89-L175). Please instead make the necessary modifications to DLC2NWB in a PR and import those functions here.

Reply:
Agreed, a very simple issue+PR would simply request they allow an existing NWBFile object to be passed in, and if so prevent the lines https://github.com/DeepLabCut/DLC2NWB/blob/main/dlc2nwb/utils.py#L157-L162 and https://github.com/DeepLabCut/DLC2NWB/blob/main/dlc2nwb/utils.py#L169-L172 from getting called, instead returning the `nwbfile` object post-modification.

Would make the extension even better as well that way, so I can't see them objecting.

Reply:
yeah makes sense, I've created a PR https://github.com/DeepLabCut/DLC2NWB/pull/10

Reply:
In the meantime, feel free to pin to that dev branch and update the code here correspondingly so we can see what it would look like